Automattic/pocket-casts-android
# Remove custom bottom sheet behavior

## Description

Continuation of Kotlin migration from #44. This PR removes [PCBottomSheetBehavior.java](https://github.com/Automattic/pocket-casts-android/blob/61fe9bf02411dbb158ec838d2fc4834099e99a4b/modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/helper/PCBottomSheetBehavior.java) and uses default behavior built into Material library.

We copied `BottomSheetBehavior` from Material some time ago because it [had issues interacting](https://github.com/Automattic/pocket-casts-android-private/pull/1778) with `ViewPager2`. Since then the library have been updated and I did not discover any issues like between these two components.

`PCBottomSheetBehavior` was [updated](https://github.com/Automattic/pocket-casts-android-private/issues/1768) with [the changes from Material](https://github.com/material-components/material-components-android/commit/6178e7146357bbd602115bcb08194a76592d89c0) to prevent some issues. After that component wasn't updated which means it stayed behind the Material library one. 

At some point `PCBottomSheetBehavior` [it was updated](https://github.com/Automattic/pocket-casts-android-private/pull/1862) to help with dragging issues. After migration I actually encountered a small regression. Especially during the initial flinging up or scrolling up motion. Investigating it made me realize that we have other bugs unrelated to the migration. For example, peek height could be left for a moment above value of `0`. This meant that lifting the finger causes the "up next" sheet to be stuck for a short time and then disappear. I tried to showcase this in the animation below but it is not super obvious as during live interaction with the app.

![screen-20240112-152320-ezgif com-optimize(1)](https://github.com/Automattic/pocket-casts-android/assets/30936061/ddd050ee-234c-4a9c-842e-0cf8eeec2eab)

I tried a lot of different approaches but the simplest one was to remove dragging motion and change it to flinging to open the "up next" bottom sheet. It can still be closed by dragging but cannot be opened this way. Users need to either fling or tap on the top right button. I commented in the code any quirks that I found while working on this feature.

## Testing Instructions

Test `ViewPager2` interaction.

1. Play an episode with longer description and with chapters like [this one](https://pca.st/episode/53cbd091-d83f-4b16-b1af-086e2dc8b9bc).
2. Open up the player.
3. Scroll to `Details`.
4. Scroll vertically through the details.
5. Scroll to `Chapters`.
6. You can also scroll to `Bookmarks` if you have any and test if vertical scrolling works there as well.
7. Scroll back to now playing.

Test interaction between the player bottom sheet and the "up next" bottom sheet. Try to "monkey test" it. Do these things in different order. Play with the functionality and try to break it. Things to test:

* The player should open by tapping on the mini-player.
* The player should open by dragging up the mini-player.
* The player should close by dragging it down by the content.
* The player should close by dragging it down by the top bar handle.
* The player should close by flinging it down.
* The player should close by tapping an arrow in the top left corner.
* The player should close by using the system back button.

---

* "Up next" should open by flinging up on the player.
* "Up next" should open by tapping on the queue in the top right corner of the player.
* "Up next" should close by dragging it down by the content.
* "Up next" should close by dragging it down by the top bar handle.
* "Up next" should close by flinging it down.
* "Up next" should close by tapping `x` in the top left corner.
* "Up next" should close by using the system back button.

## Checklist
 
#### I have tested any UI changes...
